helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1272,7 +1280,7 @@ protected void checkLiveInstancesObservation(List<LiveInstance> liveInstances,
 
   public void shutdown() throws InterruptedException {
     stopPeriodRebalance();
-
+    stopOnDemandTimer();
```

Comment:
Shall we also stopOnDemandTimer on the FINALIZE notification? This is done for the PeriodRebalance timer.

Reply:
I noticed it too. periodicalRebalanceTimer would be canceled in Finalize and created in init time. But this time before did not do this. Maybe it is for a reason. (Not that familiar with P2P) I am afraid changing its life cycle may cause some unexpected behavior.

Reply:
In theory, if a FINALIZE notification comes, then we should stop any pending tasks. Or it will keep running even without the later INIT notification comes.
But I agree that this PR won't make it worse. We can skip that change for now.